src/routes/health/health.service.ts

```diff
@@ -0,0 +1,35 @@
+import {
+  Inject,
+  Injectable,
+  ServiceUnavailableException,
+} from '@nestjs/common';
+import { IHealthRepository } from '../../domain/health/health.repository.interface';
+import { HealthEntity } from '../../domain/health/entities/health.entity';
+import { Health, HealthStatus } from './entities/health.entity';
+import {
+  ILoggingService,
+  LoggingService,
+} from '../../logging/logging.interface';
+
+@Injectable()
+export class HealthService {
+  constructor(
+    @Inject(IHealthRepository)
+    private readonly healthRepository: IHealthRepository,
+    @Inject(LoggingService)
+    private readonly loggingService: ILoggingService,
+  ) {}
+
+  async isReady(): Promise<Health> {
+    const readiness = await this.healthRepository.isReady();
+    switch (readiness) {
+      case HealthEntity.READY:
+        return new Health(HealthStatus.OK);
+      case HealthEntity.NOT_READY:
+        throw new ServiceUnavailableException(new Health(HealthStatus.KO));
+      default:
+        this.loggingService.debug(`Readiness status ${readiness} not handled`);
```

Comment:
Going throught the `default` branch of this switch shouldn't be possible theoretically, unless the `HealthRepository` implementation changes and this is not adapted accordingly, right? I'd say in that case we could use a upper log level (`warn` at least, or even `error`), since this is a unexpected/error condition in the code, what do you think?

Reply:
> Going throught the `default` branch of this switch shouldn't be possible theoretically, unless the `HealthRepository` implementation changes and this is not adapted accordingly, right? I'd say in that case we could use a upper log level (`warn` at least, or even `error`), since this is a unexpected/error condition in the code, what do you think?

I agree. I will change it to `error` because we should handle all the enum values here 👍

Reply:
[9ec1082](https://github.com/safe-global/safe-client-gateway-nest/pull/597/commits/9ec1082eda36aff48ce4a21d952c11b7fca1991b)